internal/server/router/feature.go

```diff
@@ -0,0 +1,110 @@
+package router
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/go-chi/render"
+	"github.com/oklog/ulid/v2"
+
+	"github.com/openmeterio/openmeter/api"
+	credit_model "github.com/openmeterio/openmeter/internal/credit"
+	"github.com/openmeterio/openmeter/pkg/contextx"
+	"github.com/openmeterio/openmeter/pkg/models"
+	"github.com/openmeterio/openmeter/pkg/slicesx"
+)
+
+// Get feature, GET:/api/v1/features/{featureId}
+func (a *Router) GetFeature(w http.ResponseWriter, r *http.Request, featureId api.FeatureId) {
+	ctx := contextx.WithAttr(r.Context(), "operation", "getFeature")
+	namespace := a.config.NamespaceManager.GetDefaultNamespace()
+
+	feature, err := a.config.CreditConnector.GetFeature(ctx, namespace, featureId)
+	if err != nil {
+		if _, ok := err.(*credit_model.FeatureNotFoundError); ok {
+			models.NewStatusProblem(ctx, err, http.StatusNotFound).Respond(w, r)
+			return
+		}
+
+		a.config.ErrorHandler.HandleContext(ctx, err)
+		models.NewStatusProblem(ctx, err, http.StatusInternalServerError).Respond(w, r)
+		return
+	}
+
+	_ = render.Render(w, r, feature)
+}
+
+// List features: GET /api/v1/features
+func (a *Router) ListFeatures(w http.ResponseWriter, r *http.Request) {
+	ctx := contextx.WithAttr(r.Context(), "operation", "listFeatures")
+	namespace := a.config.NamespaceManager.GetDefaultNamespace()
+
+	features, err := a.config.CreditConnector.ListFeatures(ctx, namespace, credit_model.ListFeaturesParams{})
+	if err != nil {
+		a.config.ErrorHandler.HandleContext(ctx, err)
+		models.NewStatusProblem(ctx, err, http.StatusInternalServerError).Respond(w, r)
+		return
+	}
+	list := slicesx.Map[credit_model.Feature, render.Renderer](features, func(feature credit_model.Feature) render.Renderer {
+		return &feature
+	})
+
+	_ = render.RenderList(w, r, list)
+}
+
+// Create feature, POST: /api/v1/features
+func (a *Router) CreateFeature(w http.ResponseWriter, r *http.Request) {
```

Comment:
Do we validate somewhere to only allow meters with `MeterAggregationSum`?

Reply:
That is a very good point!